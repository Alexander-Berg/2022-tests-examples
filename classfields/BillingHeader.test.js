const _ = require('lodash');
const React = require('react');
const BillingHeader = require('./BillingHeader');
const { shallow } = require('enzyme');
const { shallowToJson } = require('enzyme-to-json');
const createContextProvider = require('autoru-frontend/mocks/createContextProvider').default;
const contextMock = require('autoru-frontend/mocks/contextMock').default;

const ContextProvider = createContextProvider(contextMock);

const DEFAULT_PROPS = {
    cost: 90,
    baseCost: 90,
    isMobile: false,
    shouldShowCostInfo: true,
    title: 'Турбо-продажа',
};

let props;

beforeEach(() => {
    props = _.cloneDeep(DEFAULT_PROPS);
});

it('правильно рисует компонент для десктопа', () => {
    props.isMobile = false;
    const wrapper = shallow(<ContextProvider><BillingHeader { ...props }/></ContextProvider>).dive();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('правильно рисует компонент для мобилки', () => {
    props.isMobile = true;
    const wrapper = shallow(<ContextProvider><BillingHeader { ...props }/></ContextProvider>).dive();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('правильно рисует компонент если оплачивается 1 сервис', () => {
    const wrapper = shallow(<ContextProvider><BillingHeader { ...props }/></ContextProvider>).dive();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('покажет скидку если baseCost больше cost', () => {
    props.baseCost = 101;
    const wrapper = shallow(<ContextProvider><BillingHeader { ...props }/></ContextProvider>).dive();
    const baseCostEl = wrapper.find('.BillingHeader__baseCost');

    expect(shallowToJson(baseCostEl)).toMatchSnapshot();
});

it('не покажет скидку если baseCost равен 0', () => {
    props.baseCost = 0;
    const wrapper = shallow(<ContextProvider><BillingHeader { ...props }/></ContextProvider>).dive();
    const subtitleEl = wrapper.find('.BillingHeader__price');

    expect(shallowToJson(subtitleEl)).toMatchSnapshot();
});

it('не покажет информацию о стоимости если не передан флаг', () => {
    props.shouldShowCostInfo = false;
    const wrapper = shallow(<ContextProvider><BillingHeader { ...props }/></ContextProvider>).dive();
    const subtitleEl = wrapper.find('.BillingHeader__subtitle');

    expect(subtitleEl).toHaveLength(0);
});
