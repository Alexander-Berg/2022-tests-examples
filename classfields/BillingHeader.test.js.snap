// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`не покажет скидку если baseCost равен 0 1`] = `
<div
  className="BillingHeader__price"
>
  <span>
    90 ₽
  </span>
</div>
`;

exports[`покажет скидку если baseCost больше cost 1`] = `
<span
  className="BillingHeader__baseCost"
>
  101 ₽
</span>
`;

exports[`правильно рисует компонент для десктопа 1`] = `
<div
  className="BillingHeader BillingHeader_platform_desktop"
>
  <div
    className="BillingHeader__title"
  >
    Турбо-продажа
  </div>
  <div
    className="BillingHeader__price"
  >
    <span>
      90 ₽
    </span>
  </div>
</div>
`;

exports[`правильно рисует компонент для мобилки 1`] = `
<div
  className="BillingHeader BillingHeader_platform_mobile"
>
  <div
    className="BillingHeader__title"
  >
    Турбо-продажа
  </div>
  <div
    className="BillingHeader__price"
  >
    <span>
      90 ₽
    </span>
  </div>
</div>
`;

exports[`правильно рисует компонент если оплачивается 1 сервис 1`] = `
<div
  className="BillingHeader BillingHeader_platform_desktop"
>
  <div
    className="BillingHeader__title"
  >
    Турбо-продажа
  </div>
  <div
    className="BillingHeader__price"
  >
    <span>
      90 ₽
    </span>
  </div>
</div>
`;
